Source/JavaScriptCore/builtins/JSIteratorHelperPrototype.js

```diff
@@ -51,7 +51,12 @@ function return()
     var state = @getGeneratorInternalField(generator, @generatorFieldState);
     if (state === @GeneratorStateInit) {
         @putGeneratorInternalField(generator, @generatorFieldState, @GeneratorStateCompleted);
-        @iteratorGenericClose(@getIteratorHelperInternalField(this, @iteratorHelperFieldUnderlyingIterator));
+
+        var underlyingIterators = @getIteratorHelperInternalField(this, @iteratorHelperFieldUnderlyingIterators);
+        if (underlyingIterators) {
+            for (var i = 0; i < underlyingIterators.length; ++i)
```

Comment:
Please allow me a business day or two to think about this!

Reply:
@dcrousso Please note that https://tc39.es/proposal-joint-iteration/#sec-closeall, apart from going in reverse order, is carefully delaying abrupt completions of `return` lookups / calls from throwing an exception until all iterators are closed. It's not easy to implement in JS.

Let's take simpler approach here by just null-checking the `@getIteratorHelperInternalField(this, @iteratorHelperFieldUnderlyingIterators)`.